client/service-container-chain/src/service.rs

```diff
@@ -0,0 +1,499 @@
+        task_manager: &mut node_builder.task_manager,
+        da_recovery_profile: if collation_params.is_some() {
+            DARecoveryProfile::Collator
+        } else {
+            DARecoveryProfile::FullNode
+        },
+        import_queue: import_queue_service,
+        relay_chain_slot_duration,
+        recovery_handle: Box::new(overseer_handle.clone()),
+        sync_service: node_builder.network.sync_service.clone(),
+    })?;
+
+    if let Some(crate::spawner::CollationParams {
+        collator_key,
+        orchestrator_tx_pool,
+        orchestrator_client,
+    }) = collation_params
+    {
+        let node_spawn_handle = node_builder.task_manager.spawn_handle().clone();
+        let node_client = node_builder.client.clone();
+        let node_backend = node_builder.backend.clone();
+
+        start_consensus_container(
+            node_client.clone(),
+            node_backend.clone(),
+            orchestrator_client.clone(),
+            orchestrator_tx_pool.clone(),
+            block_import.clone(),
+            prometheus_registry.clone(),
+            node_builder.telemetry.as_ref().map(|t| t.handle()).clone(),
+            node_spawn_handle.clone(),
+            relay_chain_interface.clone(),
+            orchestrator_chain_interface.clone(),
+            node_builder.transaction_pool.clone(),
+            node_builder.network.sync_service.clone(),
+            keystore.clone(),
+            force_authoring,
+            relay_chain_slot_duration,
+            para_id,
+            orchestrator_para_id,
+            collator_key.clone(),
+            overseer_handle.clone(),
+            announce_block.clone(),
+        );
+    }
+
+    node_builder.network.start_network.start_network();
+
+    Ok((
+        node_builder.task_manager,
+        node_builder.client,
+        node_builder.backend,
+    ))
+}
+
+pub fn container_chain_import_queue(
+    parachain_config: &Configuration,
+    node_builder: &NodeBuilder<ContainerChainNodeConfig>,
+) -> (ContainerChainBlockImport, BasicQueue<Block>) {
+    // The nimbus import queue ONLY checks the signature correctness
+    // Any other checks corresponding to the author-correctness should be done
+    // in the runtime
+    let block_import =
+        ContainerChainBlockImport::new(node_builder.client.clone(), node_builder.backend.clone());
+
+    let import_queue = nimbus_consensus::import_queue(
+        node_builder.client.clone(),
+        block_import.clone(),
+        move |_, _| async move {
+            let time = sp_timestamp::InherentDataProvider::from_system_time();
+
+            Ok((time,))
+        },
+        &node_builder.task_manager.spawn_essential_handle(),
+        parachain_config.prometheus_registry(),
+        false,
+    )
+    .expect("function never fails");
+
+    (block_import, import_queue)
+}
+
+#[sc_tracing::logging::prefix_logs_with(container_log_str(para_id))]
+fn start_consensus_container(
```

Comment:
We are targeting this to be launched by both collators & non-collators right? If that is the case I think the first thing we should do is identify which things are needed by collators and which ones are not. One example is the `orchestrator_tx_pool`, which to my understanding is only needed in collation mode, as well as the `orhcestrator_para_id` or the `collator_key`

Reply:
All those things could be made optional and only pass them in collation-mode

Reply:
Sorry this was more for `start_node_impl_container`. But same logic applies. We are injecting orchestrator_para_id there when we dont need in `non-collation-mode`, so we probably should make it optional